src/signing.rs

```diff
@@ -183,7 +184,10 @@ impl SigningKey {
     /// The standard hash function used for most ed25519 libraries is SHA-512,
     /// which is available with `use sha2::Sha512` as in the example above.
     /// Other suitable hash functions include Keccak-512 and Blake2b-512.
-    #[cfg(feature = "rand_core")]
+    ///
+    /// # Features
+    ///
+    /// Requires optional feature `rand_core` activated
```

Comment:
the crate is set up to allow this:

```suggestion
#[cfg_attr(docsrs, doc(cfg(feature = "rand_core")))]
#[cfg(feature = "rand_core")]
```

Reply:
Wouldn't it be good to move to `doc_auto_cfg` instead?

Reply:
Maybe; my personal view is to wait for this to be stable before using it given what I'm reading on the tracking issue. Either way, using the built-in attributes should be preferred over doc comments.

Reply:
Can you speak to those issues in particular? Reading the list linked here:

https://github.com/rust-lang/rust/pull/100883#issuecomment-1317684866

...it seems like if `doc_auto_cfg` generates the attributes correctly, barring some other nightly regression (which won't retroactively impact previous successful docs.rs builds) it should just be a matter of migrating to `#![doc(auto_cfg)]` when the feature stabilizes.

Reply:
The gate is at the top not at the fn. The documentation documents the feature-gate that is all.

Reply:
we setup rand_core in the features for the docsrs in the metadata. the docsrs changes are in another PR